app/src/components/LayoutHeader.vue

```diff
@@ -42,30 +42,138 @@
+          </div>
+          <figure>
+            <Jazzicon :address="userAddress" :key="userAddress" width="40" />
+          </figure>
+          <div>
+            <ArrowBottomIcon class="icon-small icon-primary" />
+          </div>
+        </div>
+        <!-- menu-->
+        <div
+          class="
+            absolute
+            hidden
+            group-hover:flex
+            z-10
+            border border-grey-400
+            p-5
+            right-2
+            md:right-20
+            bg-white
+            text-grey-400
+            flex-col
+            gap-y-2
+            uppercase
+            font-medium
+            text-left
+            whitespace-nowrap
+          "
+        >
+          <div>{{ userDisplayName }}</div>
+          <div>{{ balance }} ETH</div>
+          <div class="border-b border-grey-400 my-4"></div>
+
+          <button
+            @click="changeWallet"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            change wallet
+          </button>
+          <button
+            @click="disconnect"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            disconnect wallet
+          </button>
+        </div>
       </div>
-    </nav>
+    </div>
+    <div v-else-if="!isSupportedNetwork" class="ml-auto flex items-center">
+      <div class="flex h-12 items-center pr-2 w-12">
+        <WarningIcon class="icon stroke-pink" />
+      </div>
+      <div class="text-gray-500">Unsupported network</div>
+    </div>
+    <div v-else @click="connectWallet" class="flex items-center h-14 gap-x-2 group cursor-pointer ml-auto">
+      <div class="hidden md:block group-hover:text-grey-500">Connect</div>
+      <div>
+        <ConnectWalletIcon class="icon icon-primary" />
+      </div>
+    </div>
   </header>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import { WarningIcon } from '@fusion-icons/vue/interface';
+import { Wallet3Icon as ConnectWalletIcon } from '@fusion-icons/vue/web3';
+import { ArrowBottom2Icon as ArrowBottomIcon } from '@fusion-icons/vue/interface';
+import Jazzicon from 'src/components/Jazzicon.vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { formatEther, JsonRpcProvider, Web3Provider } from 'src/utils/ethers';
 
 // Header menu bar items
 const navigation = [
   { name: 'Home', href: '/' },
   { name: 'Cart', href: '/cart' },
-  { name: 'Contact', href: '/contact' },
 ];
 
 export default defineComponent({
   name: 'LayoutHeader',
-  components: { WarningIcon },
+  components: { ConnectWalletIcon, ArrowBottomIcon, CartIcon, Jazzicon, WarningIcon },
   emits: ['toggleAbout'],
+  data() {
+    return {
+      balance: '0',
+      provider: null as Web3Provider | JsonRpcProvider | null,
+    };
+  },
+  watch: {
+    userAddress: 'updateBalance',
+  },
```

Comment:
Instead of a watcher and async method, this is part of the wallet state, so storing their balance should probably be managed in the `wallet.ts` store. However, IMO this should be removed so before making this change let's see what the decision is around that